pydantic/deprecated/tools.py

```diff
@@ -4,20 +4,26 @@
 import warnings
 from typing import Any, Callable, Type, TypeVar, Union
 
-from .. import AnalyzedType
-
-__all__ = 'parse_obj_as', 'schema_of', 'schema_json_of'
+from typing_extensions import deprecated
 
+from ..analyzed_type import AnalyzedType
 from ..json_schema import DEFAULT_REF_TEMPLATE, GenerateJsonSchema
 
+__all__ = 'parse_obj_as', 'schema_of', 'schema_json_of'
+
 NameFactory = Union[str, Callable[[Type[Any]], str]]
 
 
 T = TypeVar('T')
 
 
+@deprecated('parse_obj_as is deprecated. Use pydantic.AnalyzedType.validate_python instead.')
 def parse_obj_as(type_: type[T], obj: Any, type_name: NameFactory | None = None) -> T:
-    # TODO: add deprecation warning of some sort
+    warnings.warn(
+        'parse_obj_as is deprecated. Use pydantic.AnalyzedType.validate_python instead.',
+        DeprecationWarning,
+        stacklevel=2,
+    )
     if type_name is not None:  # pragma: no cover
```

Comment:
Should I remove this warning?

Reply:
I think it's okay to leave it. If you choose to ignore the first warning we still want the second one. We might even want it to be an error.